### PR TITLE
Check for images added-and-modified in a same PR

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

... as is currently the case in #21562.

As in #21359, this currently contains two additional commits at the top that should cause CI to fail; I will edit these commits out of existence once this PR is approved.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: .github/workflows/clean_pr.yml [hunk 1]
```diff
@@ -19,4 +19,13 @@ jobs:
             printf 'The following files were both added and deleted in this PR:\n%s\n' "$ad"
             exit 1
           fi
-
+      - name: Check for added-and-modified images
+          git fetch --quiet origin "$GITHUB_BASE_REF"
```
Comment (QuLogic) [inline .github/workflows/clean_pr.yml:23]: You forgot the `run: |`; the workflow didn't run at all.

Comment (anntzer) [reply .github/workflows/clean_pr.yml:24]: woops.

File: .github/workflows/clean_pr.yml [hunk 2]
```diff
@@ -19,4 +19,14 @@ jobs:
...
-
+      - name: Check for added-and-modified images
+        run: |
+          git fetch --quiet origin "$GITHUB_BASE_REF"
+          base="$(git merge-base "origin/$GITHUB_BASE_REF" 'HEAD^2')"
+          am="$(git log "$base..HEAD^2" --pretty=tformat: --name-status --diff-filter=AM |
+                cut --fields 2 | sort | uniq --repeated |
+                grep -E '.(png|pdf|ps|eps|svg)')"
+          if [[ -n "$am" ]]; then
+            printf 'The following images were both added and modified in this PR:\n%s\n' "$ad"
```
Comment (timhoffm) [inline .github/workflows/clean_pr.yml:30]:
```suggestion
            printf 'The following images were both added and modified in this PR:\n%s\n' "$am"
```
Unless this is some bash magic my naive bash understanding is not aware of.

Comment (anntzer) [reply .github/workflows/clean_pr.yml]: re-woops.

**Review by QuLogic — APPROVED** [review]

**Review by dstansby — CHANGES_REQUESTED** [review]
This is an approval, but I'll stick request changes on it so it doesn't accidentally get merged before you remove the test figure.

**Comment by jklymak** [comment]
@anntzer feel free to self merge when you have the image squashed out....

**Pull request merged**